compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3466,6 +3467,23 @@ boolean checkLiteralAssignabilityBasedOnType(BLangLiteral baseLiteral, BLangLite
         return false;
     }
 
+    boolean validateFloatLiteral(Location pos, String numericLiteral) {
```

Comment:
```suggestion
    boolean isValidFloatLiteral(Location pos, String numericLiteral) {
```

Reply:
Since we are logging errors also in this method (as opposed to just a check), we can use `validateFloatLiteral` right?